### PR TITLE
Use ES2015 classes instead of having prototype-based functions

### PR DESCRIPTION
**Pull request by tomasko126**
This change will practically improve readability of the code.
An algorithm for showing tabs in Options has also been reworked, which as a result improves performance.

@kpeckett Could you please review the changes I've made? Thanks 😀

### CODE REVIEW

**Comment by tomasko126** [comment]
@kpeckett I have just figured out, that I might have missed to add some arguments to some methods as in an example below.

As you can see, there is missing an `enabled` argument in brackets following after `setEnabled`.
snimka obrazovky 2016-09-07 o 9 54 39

I will compare the code from master branch and this branch and will let you know in advance, when you can start reviewing this PR.

File: catblock/channels.js [hunk 1]
```diff
@@ -136,70 +141,66 @@ Channels.prototype = {
...
-    _getLatestListings: function() {
+    _getLatestListings() {
         throw new Error("Implemented by subclass. Call callback with up-to-date listings.");
     }
-};
-
+}
 
 // Channel containing hard coded cats loaded from disk.
 // Subclass of Channel.
```
Comment (itskdog) [inline catblock/channels.js]: Second comment not needed as it is now explicit in the code.

File: catblock/channels.js [hunk 2]
```diff
@@ -238,23 +239,26 @@ AprilFoolsCatsChannel.prototype = {
             L(109, 385, "tall6.jpg")
         ]);
     }
-};
+
+}
 
 
 // Abstract base class for Flickr-based Channels.
 // Subclass of Channel.
```
Comment (itskdog) [inline catblock/channels.js]: Also here as well

File: catblock/channels.js [hunk 3]
```diff
@@ -301,55 +305,51 @@ FlickrChannel.prototype = {
         }
         return result;
     }
-};
+}
 
 // Channel pulling from Flickr search results.
 // Subclass of FlickrChannel.
```
Comment (itskdog) [inline catblock/channels.js]: And again

File: catblock/channels.js [hunk 4]
```diff
@@ -301,55 +305,51 @@ FlickrChannel.prototype = {
...
         var that = this;
         this._api("flickr.photos.search", { text: this._query }, function(resp) {
             callback(that._toListings(resp.photos));
         });
     }
-};
+}
 
 // Channel pulling from a Flickr photoset.
 // Subclass of FlickrChannel.
```
Comment (itskdog) [inline catblock/channels.js]: I'll just let you find the rest. Do a `git grep subclass` or something.

File: filtering/myfilters.js [hunk 5]
```diff
@@ -5,924 +5,924 @@
...
+        }
+
+        if (forceFetch) {
+            delete this._subscriptions[id].last_modified;
+        }
+
         if (this._subscriptions[id].subscribed) {
-            //if forceFetch, set the last update timestamp of the malware to zero, so it's updated now.
-            if (forceFetch) {
-                this._subscriptions.malware.last_update = 0;
```
Comment (itskdog) [inline filtering/myfilters.js:376]: Was this completely removed? I can't tell where it's moved to.

Comment (tomasko126) [reply filtering/myfilters.js:376]: @kpeckett Nope, it wasn't.

File: catblock/channels.js (hunk 4 above)
Comment (tomasko126) [reply catblock/channels.js]: @kpeckett I've removed all those comments.

**Pull request merged**